lang/rust/Portfile

```diff
@@ -384,3 +403,445 @@ subport rust-src {
         system -W ${rust_source_dir} "find . -type d -name test -print0 | xargs -0 rm -rf"
     }
 }
+
+# The root cause of offline mode is prevent cargo to check signatures
+# which is broken before 10.8
+# See: https://github.com/rust-lang/cargo/issues/4813
+# See: https://github.com/mozilla/releases-comm-central/commit/5cd3595203de78a6f47324f53f0cf616f4e383ea
+if {${subport} ne "${ccwrap}" } {
+    PortGroup           cargo_fetch 1.0
+
+    dist_subdir         $name
+
+    post-extract {
+        # enable offline mode for cargo at bootstrap
+        system -W ${worksrcpath} "patch -p0 < ${filespath}/offline-bootstrap.patch"
+    }
+
+    if { ${os.platform} eq "darwin" && ${os.major} < 12 } {
+        # I don't need whole libcxx
+        # anyway, it is an easy way to get some missed symbols from compiler-rt
+        legacysupport.use_mp_libcxx yes
+
+        # well, let me build a small library which contains only needed symbols
+        pre-build {
+            set export_list_file ${workpath}/scalpel.export_list
+            set scalpel_dylib ${workpath}/scalpel.dylib
+            set export_list [open ${export_list_file} w]
+
+            # Since https://github.com/rust-lang/rust/commit/8a6ff90a3a41e6ace18aeb089ea0a0eb3726dd08
+            puts ${export_list} "___muloti4"
+
+            close ${export_list}
+
+            system "${configure.cc} -dynamiclib -exported_symbols_list ${export_list_file} -o ${scalpel_dylib} ${prefix}/lib/libcxx/libc++.a"
+
+            build.env-append DYLD_FORCE_FLAT_NAMESPACE=1 \
+                        DYLD_INSERT_LIBRARIES=${scalpel_dylib}
```

Comment:
I think it would be nice if maybe in the future this `___muloti4` could be factored into say legacysupport in some way, just to avoid the need to build a small helper lib here...

Reply:
@cjones051073 well, it has a few mul options which can be extracted as separated library. Honestly, I feel that the right things to do is keep llvm's compiler-rt as dedicated port and use it when someone needed.

Reply:
what I mean is if what is done here is something other ports could benefit from, then we should avoid each port having to build the library themselves.

Reply:
Seems I figured out how to build this library after all. I'll include it inside this PR.

Reply:
@cjones051073 well, I have no idea how to make such port.

A naive approach:
```
clang -dynamiclib -Wl,-all_load -o /tmp/test.dylib /opt/local/libexec/llvm-11/lib/clang/11.1.0/lib/darwin/libclang_rt.osx.a
```
fails with duplicated symbols
```
duplicate symbol '___bswapdi2' in:
    /opt/local/libexec/llvm-11/lib/clang/11.1.0/lib/darwin/libclang_rt.osx.a(bswapdi2.c.o)
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/lib/darwin/libclang_rt.osx.a(bswapdi2.c.o)
duplicate symbol '___bswapsi2' in:
    /opt/local/libexec/llvm-11/lib/clang/11.1.0/lib/darwin/libclang_rt.osx.a(bswapsi2.c.o)
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/lib/darwin/libclang_rt.osx.a(bswapsi2.c.o)
```

Can I use `${prefix}/libexec/llvm-11/lib/clang/11.1.0/lib/darwin/libclang_rt.osx.a` as donor instead of `${prefix}/lib/libcxx/libc++.a`? I don't like because I need to hardcore llvm version also here and current approach seems better for me.

Maybe it is a good idea to move this hack with creating a dynamic library and load it to legacysupport but I have no idea how to do it on the right way.

Reply:
The usual way to make symbols from compiler_rt available to a build, when needed, is to directly link in the static library, as you mentioned. It would be probably much better to do this, if you can add static libraries to a cargo/rust build (I'm sure you must be able to) rather than to make a dylib and insert it. It would also not be a very good idea (IMHO) to add random comiler_rt symbols to the legacysupport.dylib, as -- they shouldn't need to be there.

Reply:
@kencu the issue that I need it to run `rustc` which was prebuilt. `rustc` tries to looks for `___muloti4` inside its std library which is also ships as prebuilt binary.

The only way is create a small dynamic library which I've forced to load and switched everything to flat namespace.

Honest, I doubt that such hack is needed for any another port.

Reply:
Yes, you're right. I forgot for a moment that this was needed to get the downloaded prebuilt binary to work. Right.